app/controllers/api/repositories_controller.rb

```diff
@@ -45,6 +46,22 @@ def shields_dependencies
     render json: json_hash
   end
 
+  def sync
+    if @repository.recently_synced?
+      StructuredLog.capture(
+        "REPOSITORY_SYNC_REQUEST_SKIPPED", {
+          host_type: @repository.host_type,
+          full_name: @repository.full_name,
+          repository_last_synced_at: @repository.last_synced_at,
+        }
+      )
+      render json: { error: "Repository has already been synced recently" }
+    else
+      @repository.manual_sync
```

Comment:
Could you add a corresponding log message for when we trigger the manual sync. It probably makes more sense to do that in `Repository#manual_sync` than here. We could capture that data looking through API requests, but would be easier to get a sense of how often we are doing manual syncs per repository with a direct log message.

Reply:
I've done that, and added one for `Project#manual_sync` too